### PR TITLE
E2E added to gateway

### PR DESCRIPTION
**Pull request by AleF83**

### CODE REVIEW

File: services/secure-gateway/config/gateway.json [hunk 1]
```diff
@@ -6,21 +6,23 @@
     },
     "security": {
         "enforce": true,
-        "policy_repository": {
-            "upstream_url": "ssh://git@git/tweek/repo",
-            "casbin_policy": "./model.csv",
-            "casbin_model": "./policy.conf"
+        "policyStorage": {
+            "casbinModel": "/model.conf",
+            "casbinPolicy": "policy.csv"
```
Comment (Yshayy) [inline services/secure-gateway/config/gateway.json]: remove casbinPolicy.csv (unless it's for appending default policy, and that maybe done with env variables)

File: services/secure-gateway/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (Yshayy) [inline services/secure-gateway/package.json]: maybe use make file instead

File: services/secure-gateway/policyStorage/policyStorage.go [hunk 3]
```diff
@@ -100,10 +101,15 @@ func New(workDir string, config *appConfig.PolicyStorage) (result persist.Adapte
 		log.Panicln("Minio bucket with casbin policies doesn't exist")
 	}
 
-	filePath := path.Join(workDir, config.CasbinModel)
-	error := minioClient.FGetObject(config.MinioBucketName, config.CasbinModel, filePath, minio.GetObjectOptions{})
+	filePath := path.Join(workDir, config.CasbinPolicy)
```
Comment (Yshayy) [inline services/secure-gateway/policyStorage/policyStorage.go]: use policy library, listen from event from nats in the future

File: services/secure-gateway/proxy/proxy.go [hunk 4]
```diff
@@ -18,14 +17,16 @@ func New(upstream *url.URL, token security.JWTToken) negroni.HandlerFunc {
 	if err != nil {
 		panic(fmt.Sprintf("Failed to setup request forwarding %v", err))
 	}
-	buf, err := buffer.New(fwd, buffer.Retry("IsNetworkError() && Attempts() < 3"))
+	/*buf, err := buffer.New(fwd, buffer.Retry(`IsNetworkError() && Attempts() < 3`))
```
Comment (Yshayy) [inline services/secure-gateway/proxy/proxy.go]: remove policy, maybe rethink it in the future

File: services/secure-gateway/policyStorage/policyStorage.go (hunk 3 above)
Comment (AleF83) [reply services/secure-gateway/policyStorage/policyStorage.go]: Added task #756

File: e2e/integration/utils/clients.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ nconf
   .defaults({
     AUTHORING_URL: 'http://authoring.localtest.me:4009',
     API_URL: 'http://api.localtest.me:4009',
-    MANGEMENT_URL: 'http://management.localtest.me:4009',
+    MANAGEMENT_URL: 'http://management.localtest.me:4009',
```
Comment (Yshayy) [inline e2e/integration/utils/clients.js]: Remove management reference from gateway, it's not needed

File: e2e/integration/utils/getToken.js [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
+const { promisify } = require('util');
 const jwtSign = promisify(jwt.sign);
 const readFile = promisify(fs.readFile);
 
 const jwtOptions = {
   algorithm: 'RS256',
   issuer: 'tweek',
-  expiresIn: '15m'
+  expiresIn: '15m',
+  keyid: 'test-tweek-keyid',
```
Comment (Yshayy) [inline e2e/integration/utils/getToken.js]: Remote test-tweek-keyid, not needed.
Separate flows of openid-connect and Tweek issuer

Comment (AleF83) [reply e2e/integration/utils/getToken.js]: Added issue #757

File: services/secure-gateway/router.go [hunk 7]
```diff
@@ -38,7 +38,7 @@ func NewRouter(configuration *appConfig.Configuration) Router {
 	legacyNonV1Router := mainRouter.MatcherFunc(func(r *http.Request, m *mux.RouteMatch) bool {
 		path := r.URL.EscapedPath()
 		host := strings.Split(r.Host, ":")[0]
-		result := strings.HasPrefix(path, "/health") || strings.HasPrefix(path, "/api/swagger.json") || anyString(host, configuration.V1Hosts.Authoring)
+		result := strings.HasPrefix(path, "/version") || strings.HasPrefix(path, "/health") || strings.HasPrefix(path, "/api/swagger.json") || anyString(host, configuration.V1Hosts.Authoring)
```
[the remark this answers is not in the document]
Comment (AleF83) [reply services/secure-gateway/router.go]: Added issue #758

**Pull request merged**